File: image-crud/src/main/java/org/judexmars/imagecrud/config/minio/MinioProperties.java
package org.judexmars.imagecrud.config.minio;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import org.springframework.boot.context.properties.ConfigurationProperties;

/**
 * Properties specific to Minio.
 */
@Getter
@Setter
@ConfigurationProperties(prefix = "minio")
@ToString
public class MinioProperties {

  private String url;

  private int port;

  private String accessKey;

  private String secretKey;

  private boolean secure;

  private String mainBucket;

  private String minorBucket;

  private long imageSize;

  private int ttl;
}

File: image-processor/src/main/kotlin/org/judexmars/imageprocessor/config/minio/MinioProperties.kt
package org.judexmars.imageprocessor.config.minio

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "minio")
data class MinioProperties(
    val url: String,
    val accessKey: String,
    val secretKey: String,
    val mainBucket: String,
    val minorBucket: String,
    val imageSize: Long,
)
